src/iop/spots.c

```diff
@@ -164,24 +164,38 @@ static void _resynch_params(struct dt_iop_module_t *self)
 }
 
 
-static void _reset_form_creation(GtkWidget *widget, dt_iop_module_t *self)
+static gboolean _reset_form_creation(GtkWidget *widget, dt_iop_module_t *self)
 {
   dt_iop_spots_gui_data_t *g = (dt_iop_spots_gui_data_t *)self->gui_data;
-  if(gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(g->bt_path)) ||
-      gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(g->bt_circle)) ||
-      gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(g->bt_ellipse)))
+
+  // we check the nb of shapes limit
+  dt_masks_form_t *grp = dt_masks_get_from_id(self->dev, self->blend_params->mask_id);
+  guint nb = 0;
+  if(grp && (grp->type & DT_MASKS_GROUP)) nb = g_list_length(grp->points);
+
+  if(nb >= 64)
+  {
+    dt_control_log(_("Spot module is limited to 64 shapes. Please add a new instance !"));
```

Comment:
Just spotted this... "Spot module..." Spot should have a lowercase S at start. Likewise for "Please".

Reply:
oups